.buildkite/pipeline.yml

```diff
@@ -23,34 +23,54 @@ steps:
             # Allow WP-CLI to be run as root, otherwise it throws an exception.
             # Reference: https://git.io/J9q2S
             - "WP_CLI_ALLOW_ROOT=true"
+      - peakon/git-shallow-clone#v0.0.1:
+          depth: 1
     command: |
         source /root/.bashrc
 
+        echo "--- :node: Setup Node environment"
         nvm install && nvm use
+
+        echo "--- :npm: Install Node dependencies"
         npm ci --unsafe-perm --prefer-offline --no-audit --no-progress
 
+        echo "--- :package: Run bundle prep work"
         npm run prebundle:js
```

Comment:
For reference and future optimization avenues, this is the command that takes up the bulk of the execution time.